plugins/woocommerce/tests/e2e-pw/tests/shopper/cart-checkout-restricted-coupons.spec.js

```diff
@@ -654,11 +645,16 @@ test.describe( 'Cart & Checkout Restricted Coupons', () => {
 			.getByLabel( 'Email address' )
 			.first()
 			.fill( 'marge@example.com' );
-		await page.getByRole( 'button', { name: 'Place order' } ).click();
+
+		await page
+			.getByRole( 'link', { name: 'Click here to enter your code' } )
+			.click();
+		await page.getByPlaceholder( 'Coupon code' ).fill( 'email-restricted' );
+		await page.getByRole( 'button', { name: 'Apply coupon' } ).click();
 
 		await expect(
 			page.getByText(
-				'Sorry, it seems the coupon "email-restricted" is not yours - it has now been removed from your order.'
+				'Please enter a valid email on the Checkout to use the coupon code "email-restricted".'
```

Comment:
If we change the error message for the Checkout, we will need to update this message.

Reply:
Yes, that test has been modified in flow and msg string to adjust to the new changes. Ideally, in a PHP Unit test for example, we would be fetching the string (and doing the mods we needed) directly from the PHP constant. That way, you change the message, and tests are always up to date. Unfortunately, we don't have that option easily here.